services/QuillLMS/spec/services/classlink_integration/districts_importer_spec.rb

```diff
@@ -4,136 +4,65 @@
 
 RSpec.describe ClasslinkIntegration::DistrictsImporter do
   let(:importer) { described_class.new }
+  let(:valid_response) do
+    {
+      'applications' => [
+        {
+          'tenant_id' => 456
+        },
+        {
+          'tenant_id' => 999
+        }
+      ]
+    }
+  end
 
   describe '#run' do
-    let(:proxy_url) { 'https://example.com' }
-    let(:bearer_token) { 'test_token' }
-    let(:first_district_id) { '123' }
-    let(:valid_response) do
-      {
-        'applications' => [
-          {
-            'oneroster_application_id' => first_district_id,
-            'bearer' => 'district_token',
-            'tenant_name' => 'Test District',
-            'name' => 'Test App',
-            'tenant_id' => 456,
-            'application_id' => '456',
-            'enabled' => 'true'
-          },
-          {
-            'oneroster_application_id' => '789',
-            'bearer' => 'district_token_2',
-            'tenant_name' => 'Another District',
-            'name' => 'Another App',
-            'tenant_id' => 999,
-            'application_id' => '999',
-            'enabled' => 'true'
-          }
-        ]
-      }
-    end
-    let(:district_importer) { double(run: nil) }
-
     before do
-      stub_const('ClasslinkIntegration::DistrictsImporter::PROXY_URL', proxy_url)
-      allow(HTTParty).to receive(:get).and_return(
-        valid_response
-      )
-    end
-
-    it 'processes each valid district' do
-      instance_double = double('ClasslinkInstance', district_id: 1, tenant_id: 456)
-      expect(importer).to receive(:valid_district?).twice.and_return(true)
-      expect(described_class).to receive(:find_or_create_district_and_instance).twice.and_return(instance_double)
-      expect(ClasslinkIntegration::DistrictImporter).to receive(:run).twice
-      importer.run
+      allow(ClasslinkIntegration::Client).to receive(:get).and_return(valid_response)
+      allow(ENV).to receive(:fetch).and_return('456,999')
```

Comment:
I don't think this is strictly necessary, but I think this `allow` call is more communicative of expectations if it has a `with` call attached so that we can understand which key this is intended to impact.  Alternatively, this might be a good use-case for `stub_const('ENV', { key => '456,999' })`

Given that this value changes in a number of different contexts, you might even generalize this slightly:
```ruby
let(:env_tenant_ids) { '456,999' }

before do
  stub_const('ENV', { key => env_tenant_ids })
end
```
Which would allow you to just use `let(:env_tenant_ids) { '123' }` or whatever to change this out rather than duplicating the stub logic.